shared/torngit/github.py

```diff
@@ -676,47 +678,44 @@ def _parse_response(self, res: Response):
                 )
                 return res.text
 
-    def _possibly_mark_current_installation_as_rate_limited(
+    def _possibly_mark_current_entity_as_rate_limited(
         self,
         *,
         reset_timestamp: Optional[str] = None,
         retry_in_seconds: Optional[int] = None,
+        # Couldn't type this to Token cause there's a circular import. Suggestions are welcome
+        token,
     ) -> None:
-        current_installation = self.data.get("installation")
-        if current_installation and current_installation.get("installation_id"):
-            installation_id = current_installation["installation_id"]
-            app_id = current_installation.get("app_id")
-            if retry_in_seconds is None and reset_timestamp is None:
-                log.warning(
-                    "Can't mark installation as rate limited because TTL is missing",
-                    extra=dict(installation_id=installation_id),
-                )
-                return
-            ttl_seconds = retry_in_seconds
-            if ttl_seconds is None:
-                # https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#handle-rate-limit-errors-appropriately
-                ttl_seconds = max(
-                    0,
-                    int(reset_timestamp) - int(datetime.now(timezone.utc).timestamp()),
-                )
-            if ttl_seconds > 0:
-                log.info(
-                    "Marking installation as rate limited",
-                    extra=dict(
-                        installation_id=installation_id,
-                        app_id=app_id,
-                        rate_limit_duration_seconds=ttl_seconds,
-                    ),
-                )
-                mark_installation_as_rate_limited(
-                    self._redis_connection, installation_id, ttl_seconds, app_id=app_id
-                )
+        entity_key_name = token.get("entity_name")
```

Comment:
Here's where I'm somewhat unsure what to do when it comes to a token that belongs to the token mappings, cause technically it wouldn't have an entity name

Reply:
hm... maybe check if there is an entity_key_name before proceeding?
We don't mark those bots as rate limited so... it should be fine right? Otherwise add keys to their tokens

```python
if entity_key_name is None:
  log.warning(
     "Can't mark entity as rate limited because TTL is missing",
  )
  return
```